data/src/emulator.js

```diff
@@ -782,17 +782,22 @@ class EmulatorJS {
             }
             this.textElem.innerText = this.localization("Download Game BIOS");
             const gotBios = (data) => {
-                this.checkCompression(new Uint8Array(data), this.localization("Decompress Game BIOS")).then((data) => {
-                    for (const k in data) {
-                        if (k === "!!notCompressedData") {
-                            this.gameManager.FS.writeFile(this.config.biosUrl.split('/').pop().split("#")[0].split("?")[0], data[k]);
-                            break;
-                        }
-                        if (k.endsWith('/')) continue;
-                        this.gameManager.FS.writeFile("/" + k.split('/').pop(), data[k]);
-                    }
+                if (this.getCore() === "same_cdi") {
+                    this.gameManager.FS.writeFile(this.config.biosUrl.split('/').pop().split("#")[0].split("?")[0], new Uint8Array(data));
                     resolve();
-                })
+                } else {
```

Comment:
No need to have an else statement here, we can return from inside the if statement since it handles the rest of the function

similar to how downloading the rom works: https://github.com/EmulatorJS/EmulatorJS/blob/9598c58cf8089833b3dfb0bd193a58ba4b77b87f/data/src/emulator.js#L848